src/main/java/com/commercetools/sync/commons/utils/QueryAll.java

```diff
@@ -107,12 +114,14 @@ private QueryAll(@Nonnull final SphereClient client,
      *         the method returns a completed future containing null.
      */
     @Nonnull
+    @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION") // `https://github.com/findbugsproject/findbugs/issues/79
     private CompletionStage<PagedQueryResult<T>> processPageAndGetNext(@Nonnull final PagedQueryResult<T> page) {
         final List<T> currentPageElements = page.getResults();
-        if (currentPageElements.size() > 0) {
+        if (!currentPageElements.isEmpty()) {
```

Comment:
Why this change?

Reply:
it was sonarlint suggestion:
> Using Collection.size() to test for emptiness works, but using Collection.isEmpty() makes the code more readable and can be more performant. The time complexity of any isEmpty() method implementation should be O(1) whereas some implementations of size() can be O(n).

https://rules.sonarsource.com/java/tag/clumsy/RSPEC-1155